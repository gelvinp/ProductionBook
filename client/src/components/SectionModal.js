import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button, Header, Confirm, Modal, Form } from 'semantic-ui-react'

class SectionModal extends Component {
  state = {
    nameField: '',
    confirmOpen: false,
  }
  closeModal = () => {
    this.setState({
      name: '',
      confirmOpen: false,
    })
    const { closeModal } = this.props
    closeModal()
  }
  componentDidUpdate(prevProps) {
    if (!prevProps.modalOpen && this.props.modalOpen) {
      const { name } = this.props
      this.setState({ nameField: name })
    }
  }
  render() {
    const { modalOpen, name } = this.props
    const { nameField, confirmOpen } = this.state
    return (
      <Modal size="tiny" open={modalOpen} onClose={this.closeModal}>
        <Modal.Header>Section Settings</Modal.Header>
        <Modal.Content>
          <Modal.Description style={{ paddingBottom: '2em' }}>
            <Form style={{ paddingBottom: '1em' }}>
              <Form.Input
                label="Name"
                width={6}
                onChange={e => this.setState({ nameField: e.target.value })}
                value={nameField}
              />
              <Button
                onClick={() => this.setState({ confirmOpen: true })}
                basic
                color="red"
              >
                Delete Section
              </Button>
              <Confirm
                open={confirmOpen}
                size="mini"
                content="This will delete all associated documents too!"
                header="Are you sure?"
                onCancel={() => this.setState({ confirmOpen: false })}
                onConfirm={() => this.setState({ confirmOpen: false })}
              />
            </Form>
            <Button basic floated="left" onClick={this.closeModal}>
              Cancel
            </Button>
            <Button primary floated="right" onClick={this.closeModal}>
              Save
            </Button>
          </Modal.Description>
        </Modal.Content>
      </Modal>
    )
  }
}

SectionModal.propTypes = {
  modalOpen: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  closeModal: PropTypes.func.isRequired,
}

export default SectionModal
