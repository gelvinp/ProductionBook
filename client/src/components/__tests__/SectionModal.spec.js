import React from 'react'
import SectionModal from '../SectionModal.js'
import { shallow, mount } from 'enzyme'

describe('SectionModal', () => {
  let modalOpen,
    name,
    closeModal,
    renameSection,
    submitRenameSection,
    submitDeleteSection,
    deleteSection,
    id

  it('Is hidden when closed', () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = mount(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    expect(wrapper.find('ModalContent').length).toBe(0)
    wrapper.unmount()
  })

  it('Is visible when open', () => {
    modalOpen = true
    name = 'test'
    closeModal = jest.fn()
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = mount(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    expect(wrapper.find('ModalContent').length).toBe(1)
    expect(wrapper).toMatchSnapshot()
    wrapper.unmount()
  })

  it('Updates the state when the section is updated', () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.find('FormInput').simulate('change', { target: { value: 'Test' } })
    expect(wrapper.state('nameField')).toEqual('Test')
  })

  it('resets state when closing', () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    const initialState = wrapper.state()
    wrapper.setState({
      nameField: 'Test',
      confirmOpen: true,
      nameFieldError: true,
      deleteError: true,
    })
    wrapper.instance().closeModal()
    expect(wrapper.state()).toEqual(initialState)
    expect(closeModal.mock.calls.length).toBe(1)
  })

  it('sets nameField when opening', () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    expect(wrapper.state().nameField).toEqual('')
    wrapper.setProps({ modalOpen: true })
    expect(wrapper.state().nameField).toEqual('test')
    wrapper.setProps({ name: 'this shouldnt update' })
    expect(wrapper.state().nameField).toEqual('test')
  })

  it('Accepts enter keydown', () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    submitRenameSection = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        error: true,
      })
    })
    renameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.setState({ nameField: 'Hello' })
    wrapper.find('FormInput').simulate('keydown', { key: 'Enter' })
    expect(submitRenameSection).toHaveBeenCalled()
  })

  it('Rejects non enter keydown', () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.setState({ nameField: 'Hello' })
    wrapper.find('FormInput').simulate('keydown', { key: 'Down' })
    expect(submitRenameSection).not.toHaveBeenCalled()
  })

  it('Rejects blank or unchanged nameField', () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.instance().renameSection()
    expect(submitRenameSection).not.toHaveBeenCalled()
    wrapper.setState({ nameField: 'test' })
    wrapper.instance().renameSection()
    expect(submitRenameSection).not.toHaveBeenCalled()
  })

  it('Shows an error on error', async () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    submitRenameSection = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        error: true,
      })
    })
    renameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.setState({ nameField: 'Hello' })
    await wrapper.instance().renameSection()
    expect(wrapper.state().nameFieldError).toBeTruthy()
  })

  it('Shows an error on non success', async () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    submitRenameSection = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        error: false,
        data: {
          success: false,
        },
      })
    })
    renameSection = jest.fn()
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.setState({ nameField: 'Hello' })
    await wrapper.instance().renameSection()
    expect(wrapper.state().nameFieldError).toBeTruthy()
  })

  it('Renames and closes when successful', async () => {
    expect.assertions(4)
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    submitRenameSection = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        error: false,
        data: {
          success: true,
        },
      })
    })
    renameSection = jest.fn().mockImplementation((id, name) => {
      expect(id).toBe(1)
      expect(name).toEqual('Hello')
    })
    submitDeleteSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.setState({ nameField: 'Hello' })
    await wrapper.instance().renameSection()
    expect(renameSection).toHaveBeenCalled()
    expect(closeModal).toHaveBeenCalled()
  })

  it('(delete) Shows an error on error', async () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    submitDeleteSection = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        error: true,
      })
    })
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.setState({ confirmOpen: true })
    await wrapper.instance().deleteSection()
    expect(wrapper.state().deleteError).toBeTruthy()
    expect(wrapper.state().confirmOpen).not.toBeTruthy()
  })

  it('(delete) Shows an error on non success', async () => {
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    submitDeleteSection = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        error: false,
        data: {
          success: false,
        },
      })
    })
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    deleteSection = jest.fn()
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper.setState({ confirmOpen: true })
    await wrapper.instance().deleteSection()
    expect(wrapper.state().deleteError).toBeTruthy()
    expect(wrapper.state().confirmOpen).not.toBeTruthy()
  })

  it('Deletes when successful', async () => {
    expect.assertions(2)
    modalOpen = false
    name = 'test'
    closeModal = jest.fn()
    submitDeleteSection = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        error: false,
        data: {
          success: true,
        },
      })
    })
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    deleteSection = jest.fn().mockImplementation(id => {
      expect(id).toBe(1)
    })
    id = 1
    const wrapper = shallow(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    await wrapper.instance().deleteSection()
    expect(deleteSection).toHaveBeenCalled()
  })

  it('Confirms before deleting', () => {
    modalOpen = true
    name = 'test'
    closeModal = jest.fn()
    renameSection = jest.fn()
    submitRenameSection = jest.fn()
    submitDeleteSection = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        error: true,
      })
    })
    deleteSection = jest.fn()
    id = 1
    const wrapper = mount(
      <SectionModal
        modalOpen={modalOpen}
        name={name}
        closeModal={closeModal}
        renameSection={renameSection}
        submitRenameSection={submitRenameSection}
        submitDeleteSection={submitDeleteSection}
        deleteSection={deleteSection}
        id={id}
      />
    )
    wrapper
      .find('#deleteSectionButton')
      .first()
      .simulate('click')
    expect(wrapper.state().confirmOpen).toBeTruthy()
    wrapper
      .find('.ui.primary.button')
      .not('.floated')
      .simulate('click')
    expect(submitDeleteSection).toHaveBeenCalled()
    wrapper
      .find('#deleteSectionButton')
      .first()
      .simulate('click')
    expect(wrapper.state().confirmOpen).toBeTruthy()
    submitDeleteSection.mockClear()
    wrapper
      .find('.ui.button')
      .not('.primary')
      .not('.floated')
      .not('#deleteSectionButton')
      .simulate('click')
    expect(submitDeleteSection).not.toHaveBeenCalled()
  })
})
