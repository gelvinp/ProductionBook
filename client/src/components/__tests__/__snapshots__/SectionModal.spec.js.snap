// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionModal Is visible when open 1`] = `
<SectionModal
  closeModal={[MockFunction]}
  deleteSection={[MockFunction]}
  id={1}
  modalOpen={true}
  name="test"
  renameSection={[MockFunction]}
  submitDeleteSection={[MockFunction]}
  submitRenameSection={[MockFunction]}
>
  <Modal
    centered={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={true}
    size="tiny"
  >
    <Portal
      closeOnDocumentClick={false}
      closeOnEscape={true}
      eventPool="Modal"
      mountNode={
        <body
          class="dimmable dimmed"
        >
          <div
            class="ui page modals dimmer transition visible active"
            style="display: flex;"
          >
            <div
              class="ui tiny modal transition visible active"
            >
              <div
                class="header"
              >
                Section Settings
              </div>
              <div
                class="content"
              >
                <div
                  class="description"
                  style="padding-bottom: 2em;"
                >
                  <form
                    class="ui form"
                    style="padding-bottom: 1em;"
                  >
                    <div
                      class="six wide field"
                    >
                      <label>
                        Name
                      </label>
                      <div
                        class="ui input"
                      >
                        <input
                          aria-invalid="true"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <h4
                      class="ui header"
                      style="display: none;"
                    >
                      An error occurred, please try again
                    </h4>
                  </form>
                  <button
                    class="ui red basic button"
                    id="deleteSectionButton"
                  >
                    Delete Section
                  </button>
                  <div
                    class="ui hidden divider"
                  />
                  <button
                    class="ui basic left floated button"
                  >
                    Cancel
                  </button>
                  <button
                    class="ui primary right floated button"
                  >
                    Save
                  </button>
                </div>
              </div>
            </div>
          </div>
        </body>
      }
      onClose={[Function]}
      onMount={[Function]}
      onOpen={[Function]}
      onUnmount={[Function]}
      open={true}
      openOnTriggerClick={true}
    >
      <PortalInner
        innerRef={
          Object {
            "current": <div
              class="ui page modals dimmer transition visible active"
              style="display: flex;"
            >
              <div
                class="ui tiny modal transition visible active"
              >
                <div
                  class="header"
                >
                  Section Settings
                </div>
                <div
                  class="content"
                >
                  <div
                    class="description"
                    style="padding-bottom: 2em;"
                  >
                    <form
                      class="ui form"
                      style="padding-bottom: 1em;"
                    >
                      <div
                        class="six wide field"
                      >
                        <label>
                          Name
                        </label>
                        <div
                          class="ui input"
                        >
                          <input
                            aria-invalid="true"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <h4
                        class="ui header"
                        style="display: none;"
                      >
                        An error occurred, please try again
                      </h4>
                    </form>
                    <button
                      class="ui red basic button"
                      id="deleteSectionButton"
                    >
                      Delete Section
                    </button>
                    <div
                      class="ui hidden divider"
                    />
                    <button
                      class="ui basic left floated button"
                    >
                      Cancel
                    </button>
                    <button
                      class="ui primary right floated button"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </div>,
          }
        }
        mountNode={
          <body
            class="dimmable dimmed"
          >
            <div
              class="ui page modals dimmer transition visible active"
              style="display: flex;"
            >
              <div
                class="ui tiny modal transition visible active"
              >
                <div
                  class="header"
                >
                  Section Settings
                </div>
                <div
                  class="content"
                >
                  <div
                    class="description"
                    style="padding-bottom: 2em;"
                  >
                    <form
                      class="ui form"
                      style="padding-bottom: 1em;"
                    >
                      <div
                        class="six wide field"
                      >
                        <label>
                          Name
                        </label>
                        <div
                          class="ui input"
                        >
                          <input
                            aria-invalid="true"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <h4
                        class="ui header"
                        style="display: none;"
                      >
                        An error occurred, please try again
                      </h4>
                    </form>
                    <button
                      class="ui red basic button"
                      id="deleteSectionButton"
                    >
                      Delete Section
                    </button>
                    <div
                      class="ui hidden divider"
                    />
                    <button
                      class="ui basic left floated button"
                    >
                      Cancel
                    </button>
                    <button
                      class="ui primary right floated button"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </body>
        }
        onMount={[Function]}
        onUnmount={[Function]}
      >
        <Portal
          containerInfo={
            <body
              class="dimmable dimmed"
            >
              <div
                class="ui page modals dimmer transition visible active"
                style="display: flex;"
              >
                <div
                  class="ui tiny modal transition visible active"
                >
                  <div
                    class="header"
                  >
                    Section Settings
                  </div>
                  <div
                    class="content"
                  >
                    <div
                      class="description"
                      style="padding-bottom: 2em;"
                    >
                      <form
                        class="ui form"
                        style="padding-bottom: 1em;"
                      >
                        <div
                          class="six wide field"
                        >
                          <label>
                            Name
                          </label>
                          <div
                            class="ui input"
                          >
                            <input
                              aria-invalid="true"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                        <h4
                          class="ui header"
                          style="display: none;"
                        >
                          An error occurred, please try again
                        </h4>
                      </form>
                      <button
                        class="ui red basic button"
                        id="deleteSectionButton"
                      >
                        Delete Section
                      </button>
                      <div
                        class="ui hidden divider"
                      />
                      <button
                        class="ui basic left floated button"
                      >
                        Cancel
                      </button>
                      <button
                        class="ui primary right floated button"
                      >
                        Save
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </body>
          }
        >
          <Ref
            innerRef={[Function]}
          >
            <RefFindNode
              innerRef={[Function]}
            >
              <div
                className="ui page modals dimmer transition visible active"
              >
                <Ref
                  innerRef={
                    Object {
                      "current": <div
                        class="ui tiny modal transition visible active"
                      >
                        <div
                          class="header"
                        >
                          Section Settings
                        </div>
                        <div
                          class="content"
                        >
                          <div
                            class="description"
                            style="padding-bottom: 2em;"
                          >
                            <form
                              class="ui form"
                              style="padding-bottom: 1em;"
                            >
                              <div
                                class="six wide field"
                              >
                                <label>
                                  Name
                                </label>
                                <div
                                  class="ui input"
                                >
                                  <input
                                    aria-invalid="true"
                                    type="text"
                                    value=""
                                  />
                                </div>
                              </div>
                              <h4
                                class="ui header"
                                style="display: none;"
                              >
                                An error occurred, please try again
                              </h4>
                            </form>
                            <button
                              class="ui red basic button"
                              id="deleteSectionButton"
                            >
                              Delete Section
                            </button>
                            <div
                              class="ui hidden divider"
                            />
                            <button
                              class="ui basic left floated button"
                            >
                              Cancel
                            </button>
                            <button
                              class="ui primary right floated button"
                            >
                              Save
                            </button>
                          </div>
                        </div>
                      </div>,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <div
                          class="ui tiny modal transition visible active"
                        >
                          <div
                            class="header"
                          >
                            Section Settings
                          </div>
                          <div
                            class="content"
                          >
                            <div
                              class="description"
                              style="padding-bottom: 2em;"
                            >
                              <form
                                class="ui form"
                                style="padding-bottom: 1em;"
                              >
                                <div
                                  class="six wide field"
                                >
                                  <label>
                                    Name
                                  </label>
                                  <div
                                    class="ui input"
                                  >
                                    <input
                                      aria-invalid="true"
                                      type="text"
                                      value=""
                                    />
                                  </div>
                                </div>
                                <h4
                                  class="ui header"
                                  style="display: none;"
                                >
                                  An error occurred, please try again
                                </h4>
                              </form>
                              <button
                                class="ui red basic button"
                                id="deleteSectionButton"
                              >
                                Delete Section
                              </button>
                              <div
                                class="ui hidden divider"
                              />
                              <button
                                class="ui basic left floated button"
                              >
                                Cancel
                              </button>
                              <button
                                class="ui primary right floated button"
                              >
                                Save
                              </button>
                            </div>
                          </div>
                        </div>,
                      }
                    }
                  >
                    <div
                      className="ui tiny modal transition visible active"
                      style={Object {}}
                    >
                      <MountNode
                        className="dimmable dimmed"
                      />
                      <ModalHeader>
                        <div
                          className="header"
                        >
                          Section Settings
                        </div>
                      </ModalHeader>
                      <ModalContent>
                        <div
                          className="content"
                        >
                          <ModalDescription
                            style={
                              Object {
                                "paddingBottom": "2em",
                              }
                            }
                          >
                            <div
                              className="description"
                              style={
                                Object {
                                  "paddingBottom": "2em",
                                }
                              }
                            >
                              <Form
                                as="form"
                                style={
                                  Object {
                                    "paddingBottom": "1em",
                                  }
                                }
                              >
                                <form
                                  className="ui form"
                                  onSubmit={[Function]}
                                  style={
                                    Object {
                                      "paddingBottom": "1em",
                                    }
                                  }
                                >
                                  <FormInput
                                    as={[Function]}
                                    control={[Function]}
                                    error={false}
                                    label="Name"
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    value=""
                                    width={6}
                                  >
                                    <FormField
                                      control={[Function]}
                                      error={false}
                                      label="Name"
                                      onChange={[Function]}
                                      onKeyDown={[Function]}
                                      value=""
                                      width={6}
                                    >
                                      <div
                                        className="six wide field"
                                      >
                                        <label>
                                          Name
                                        </label>
                                        <Input
                                          aria-describedby={null}
                                          aria-invalid={true}
                                          onChange={[Function]}
                                          onKeyDown={[Function]}
                                          type="text"
                                          value=""
                                        >
                                          <div
                                            className="ui input"
                                          >
                                            <input
                                              aria-describedby={null}
                                              aria-invalid={true}
                                              onChange={[Function]}
                                              onKeyDown={[Function]}
                                              type="text"
                                              value=""
                                            />
                                          </div>
                                        </Input>
                                      </div>
                                    </FormField>
                                  </FormInput>
                                  <Header
                                    as="h4"
                                    style={
                                      Object {
                                        "display": "none",
                                      }
                                    }
                                  >
                                    <h4
                                      className="ui header"
                                      style={
                                        Object {
                                          "display": "none",
                                        }
                                      }
                                    >
                                      An error occurred, please try again
                                    </h4>
                                  </Header>
                                </form>
                              </Form>
                              <Button
                                as="button"
                                basic={true}
                                color="red"
                                id="deleteSectionButton"
                                onClick={[Function]}
                              >
                                <Ref
                                  innerRef={
                                    Object {
                                      "current": <button
                                        class="ui red basic button"
                                        id="deleteSectionButton"
                                      >
                                        Delete Section
                                      </button>,
                                    }
                                  }
                                >
                                  <RefFindNode
                                    innerRef={
                                      Object {
                                        "current": <button
                                          class="ui red basic button"
                                          id="deleteSectionButton"
                                        >
                                          Delete Section
                                        </button>,
                                      }
                                    }
                                  >
                                    <button
                                      className="ui red basic button"
                                      id="deleteSectionButton"
                                      onClick={[Function]}
                                    >
                                      Delete Section
                                    </button>
                                  </RefFindNode>
                                </Ref>
                              </Button>
                              <Confirm
                                cancelButton="Cancel"
                                confirmButton="OK"
                                content="This will delete all associated documents too!"
                                header="Are you sure?"
                                onCancel={[Function]}
                                onConfirm={[Function]}
                                open={false}
                                size="mini"
                              >
                                <Modal
                                  centered={true}
                                  closeOnDimmerClick={true}
                                  closeOnDocumentClick={false}
                                  dimmer={true}
                                  eventPool="Modal"
                                  onClose={[Function]}
                                  open={false}
                                  size="mini"
                                >
                                  <Portal
                                    closeOnDocumentClick={false}
                                    closeOnEscape={true}
                                    eventPool="Modal"
                                    mountNode={
                                      <body
                                        class="dimmable dimmed"
                                      >
                                        <div
                                          class="ui page modals dimmer transition visible active"
                                          style="display: flex;"
                                        >
                                          <div
                                            class="ui tiny modal transition visible active"
                                          >
                                            <div
                                              class="header"
                                            >
                                              Section Settings
                                            </div>
                                            <div
                                              class="content"
                                            >
                                              <div
                                                class="description"
                                                style="padding-bottom: 2em;"
                                              >
                                                <form
                                                  class="ui form"
                                                  style="padding-bottom: 1em;"
                                                >
                                                  <div
                                                    class="six wide field"
                                                  >
                                                    <label>
                                                      Name
                                                    </label>
                                                    <div
                                                      class="ui input"
                                                    >
                                                      <input
                                                        aria-invalid="true"
                                                        type="text"
                                                        value=""
                                                      />
                                                    </div>
                                                  </div>
                                                  <h4
                                                    class="ui header"
                                                    style="display: none;"
                                                  >
                                                    An error occurred, please try again
                                                  </h4>
                                                </form>
                                                <button
                                                  class="ui red basic button"
                                                  id="deleteSectionButton"
                                                >
                                                  Delete Section
                                                </button>
                                                <div
                                                  class="ui hidden divider"
                                                />
                                                <button
                                                  class="ui basic left floated button"
                                                >
                                                  Cancel
                                                </button>
                                                <button
                                                  class="ui primary right floated button"
                                                >
                                                  Save
                                                </button>
                                              </div>
                                            </div>
                                          </div>
                                        </div>
                                      </body>
                                    }
                                    onClose={[Function]}
                                    onMount={[Function]}
                                    onOpen={[Function]}
                                    onUnmount={[Function]}
                                    open={false}
                                    openOnTriggerClick={true}
                                  />
                                </Modal>
                              </Confirm>
                              <Divider
                                hidden={true}
                              >
                                <div
                                  className="ui hidden divider"
                                />
                              </Divider>
                              <Button
                                as="button"
                                basic={true}
                                floated="left"
                                onClick={[Function]}
                              >
                                <Ref
                                  innerRef={
                                    Object {
                                      "current": <button
                                        class="ui basic left floated button"
                                      >
                                        Cancel
                                      </button>,
                                    }
                                  }
                                >
                                  <RefFindNode
                                    innerRef={
                                      Object {
                                        "current": <button
                                          class="ui basic left floated button"
                                        >
                                          Cancel
                                        </button>,
                                      }
                                    }
                                  >
                                    <button
                                      className="ui basic left floated button"
                                      onClick={[Function]}
                                    >
                                      Cancel
                                    </button>
                                  </RefFindNode>
                                </Ref>
                              </Button>
                              <Button
                                as="button"
                                floated="right"
                                onClick={[Function]}
                                primary={true}
                              >
                                <Ref
                                  innerRef={
                                    Object {
                                      "current": <button
                                        class="ui primary right floated button"
                                      >
                                        Save
                                      </button>,
                                    }
                                  }
                                >
                                  <RefFindNode
                                    innerRef={
                                      Object {
                                        "current": <button
                                          class="ui primary right floated button"
                                        >
                                          Save
                                        </button>,
                                      }
                                    }
                                  >
                                    <button
                                      className="ui primary right floated button"
                                      onClick={[Function]}
                                    >
                                      Save
                                    </button>
                                  </RefFindNode>
                                </Ref>
                              </Button>
                            </div>
                          </ModalDescription>
                        </div>
                      </ModalContent>
                    </div>
                  </RefFindNode>
                </Ref>
              </div>
            </RefFindNode>
          </Ref>
        </Portal>
      </PortalInner>
      <EventStack
        name="mouseleave"
        on={[Function]}
        pool="Modal"
        target={
          Object {
            "current": <div
              class="ui page modals dimmer transition visible active"
              style="display: flex;"
            >
              <div
                class="ui tiny modal transition visible active"
              >
                <div
                  class="header"
                >
                  Section Settings
                </div>
                <div
                  class="content"
                >
                  <div
                    class="description"
                    style="padding-bottom: 2em;"
                  >
                    <form
                      class="ui form"
                      style="padding-bottom: 1em;"
                    >
                      <div
                        class="six wide field"
                      >
                        <label>
                          Name
                        </label>
                        <div
                          class="ui input"
                        >
                          <input
                            aria-invalid="true"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <h4
                        class="ui header"
                        style="display: none;"
                      >
                        An error occurred, please try again
                      </h4>
                    </form>
                    <button
                      class="ui red basic button"
                      id="deleteSectionButton"
                    >
                      Delete Section
                    </button>
                    <div
                      class="ui hidden divider"
                    />
                    <button
                      class="ui basic left floated button"
                    >
                      Cancel
                    </button>
                    <button
                      class="ui primary right floated button"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </div>,
          }
        }
      />
      <EventStack
        name="mouseenter"
        on={[Function]}
        pool="Modal"
        target={
          Object {
            "current": <div
              class="ui page modals dimmer transition visible active"
              style="display: flex;"
            >
              <div
                class="ui tiny modal transition visible active"
              >
                <div
                  class="header"
                >
                  Section Settings
                </div>
                <div
                  class="content"
                >
                  <div
                    class="description"
                    style="padding-bottom: 2em;"
                  >
                    <form
                      class="ui form"
                      style="padding-bottom: 1em;"
                    >
                      <div
                        class="six wide field"
                      >
                        <label>
                          Name
                        </label>
                        <div
                          class="ui input"
                        >
                          <input
                            aria-invalid="true"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <h4
                        class="ui header"
                        style="display: none;"
                      >
                        An error occurred, please try again
                      </h4>
                    </form>
                    <button
                      class="ui red basic button"
                      id="deleteSectionButton"
                    >
                      Delete Section
                    </button>
                    <div
                      class="ui hidden divider"
                    />
                    <button
                      class="ui basic left floated button"
                    >
                      Cancel
                    </button>
                    <button
                      class="ui primary right floated button"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </div>,
          }
        }
      />
      <EventStack
        name="mousedown"
        on={[Function]}
        pool="Modal"
        target="document"
      />
      <EventStack
        name="click"
        on={[Function]}
        pool="Modal"
        target="document"
      />
      <EventStack
        name="keydown"
        on={[Function]}
        pool="Modal"
        target="document"
      />
    </Portal>
  </Modal>
</SectionModal>
`;
